PetrolEngine.cpp

```diff
@@ -12,9 +12,13 @@ PetrolEngine::PetrolEngine(int power, float capacity, int gears)
 
 void PetrolEngine::changeGear(int gear)
 {
-    // TODO: Add checking if gear is between -1 and gears_
-    // -1 is for REAR
-    // 0 is for NEUTRAL
-    currentGear_ = gear;
-    std::cout << __FUNCTION__ << std::endl;
+    if(-1 <= gear && gear <= PetrolEngine::gears_)
+    {
+        currentGear_ = gear;
+        std::cout << __FUNCTION__ <<" to: "<<currentGear_<< std::endl;
+    }
+    else if(gear < -1 || gears_ < gear)
+        throw std::out_of_range ("Invalid gear!");
+    else if(gear == -1 && currentGear_ > 1)
+        throw std::out_of_range ("Invalid gear! You can change to R only from 1 or N");
```

Comment:
Brakuje klasy wyjątku InvalidGear. Używacie std::out_of_range.
W tym punkcie chodziło o to, aby utworzyć taką klasę o nazwie InvalidRange i podziedziczyć ją np z std::out_of_range którego używacie.